src/ert/_c_wrappers/job_queue/ext_job.py

```diff
@@ -221,7 +230,15 @@ def get_arglist(self) -> List[str]:
         ]
 
     def get_argvalues(self) -> List[str]:
-        return list(self._get_argvalues())
+        arg_vals = []
+        for arg in self._get_arglist():
+            if self._private_context is not None:
+                arg = self._private_context.substitute(arg)
+            if self._global_context is not None:
+                arg = self._global_context.substitute(arg)
+            arg = self.get_default_mapping().get(arg, arg)
+            arg_vals.append(arg)
+        return arg_vals
```

Comment:
I assume the global context should overwrite private context. Is this related to concepts userconfig and siteconfig maybe? What i fail to understand is the the default_mapping on line 239, is there some default substitutions defined that will always be done when working on these configuration files?

Reply:
> I assume the global context should overwrite the private context.

Actually is the other way around private context will overwrite global context
example let's say you have in the private context the following substitution defined:
<Location>="Bergen"
And in the global context:
<Location>="Oslo"
default mapping
<Location>="New York"

When we encounter the following arguments: ["We", "are", "going",  "to",  "<Location>"]
for argument "<Location>" the code above will replace "<Location>" with "Bergen"
and not with "New York"

> Is this related to concepts userconfig and siteconfig maybe?
You might consider them a bit similar but the concept is not related to user config and site config. 

> What I fail to understand is  the default_mapping on line 239, is there some default substitutions defined that will always be done when working on these configuration files
So yes there can be a default mapping thing that will be used to substitute arguments from jobs but that is the last thing that happens if it is defined and something else from the private or global context has not replaced it before.

Reply:
Thanks for the clarification. After the description i see the order of substitution is top-down, and not bottom-up as i first interpreted this to be (looking at re-assignments of arg). :+1: